notebooks/k_nearest_neighbors.livemd

```diff
@@ -0,0 +1,812 @@
+        |> Vl.encode_field(:y, "density", type: :quantitative),
+        Vl.new(
+          title: [
+            text: "KDE plot of Angle of Attack feature",
+            offset: 20
+          ],
+          width: 350,
+          height: 300
+        )
+        |> Vl.data_from_values(data)
+        |> Vl.transform(density: "angle_of_attack")
+        |> Vl.mark(:area, color: "lightgreen")
+        |> Vl.encode_field(:x, "value", type: :quantitative, title: "Angle of attack")
+        |> Vl.encode_field(:y, "density", type: :quantitative),
+        Vl.new(
+          title: [
+            text: "KDE plot of Chord length feature",
+            offset: 20
+          ],
+          width: 350,
+          height: 300
+        )
+        |> Vl.data_from_values(data)
+        |> Vl.transform(density: "chord_length")
+        |> Vl.mark(:area, color: "pink")
+        |> Vl.encode_field(:x, "value", type: :quantitative, title: "Chord length")
+        |> Vl.encode_field(:y, "density", type: :quantitative)
+      ],
+      :vertical
+    ),
+    Vl.new()
+    |> Vl.concat(
+      [
+        Vl.new(
+          title: [
+            text: "KDE plot of Free Stream Velocity feature",
+            offset: 20
+          ],
+          width: 350,
+          height: 300
+        )
+        |> Vl.data_from_values(data)
+        |> Vl.transform(density: "free_stream_velocity")
+        |> Vl.mark(:area, color: "purple")
+        |> Vl.encode_field(:x, "value", type: :quantitative, title: "Free-stream velocity")
+        |> Vl.encode_field(:y, "density", type: :quantitative),
+        Vl.new(
+          title: [
+            text: ["KDE plot of Suction Side Displacement", "Thickness feature"],
+            offset: 20
+          ],
+          width: 350,
+          height: 300
+        )
+        |> Vl.data_from_values(data)
+        |> Vl.transform(density: "suction_side_displacement_thickness")
+        |> Vl.mark(:area, color: "red")
+        |> Vl.encode_field(:x, "value",
+          type: :quantitative,
+          title: "Suction Side Displacement Thickness"
+        )
+        |> Vl.encode_field(:y, "density", type: :quantitative)
+      ],
+      :vertical
+    )
+  ],
+  :horizontal
+)
+```
+
+The Suction Side Displacement Thickness and Frequency features have asymmetric distributions with many observations having small values. To make these distributions less skewed, we will use the logarithms of these two features.
+
+```elixir
+freq_log = data["frequency"] |> S.to_tensor() |> Nx.log() |> Nx.as_type(:f64)
+
+suction_side_displacement_thickness_log =
+  data["suction_side_displacement_thickness"] |> S.to_tensor() |> Nx.log() |> Nx.as_type(:f64)
+
+data = DF.discard(data, ["frequency", "suction_side_displacement_thickness"])
+data = DF.put(data, :frequency_log, freq_log)
+
+data =
+  DF.put(data, :suction_side_displacement_thickness_log, suction_side_displacement_thickness_log)
```

Comment:
This part is not elegant, but I didn't know how to avoid casting to tensors. Both `:math.log` and `Nx.log` currently don't work inside `DF.mutate`. It will be great if `Explorer` support math operations such as log, exp, etc. Should I send an issue @josevalim?

Reply:
Yes, we should support `log`. Please open up an issue!

Reply:
I added `DF.mutate`, but currently compilation of Explorer takes a lot of time, so I can revert changes and add them with a new version of Explorer.

Reply:
We will ship v0.5.6 soon!